app/src/server/utils/convertFunctionCalls.ts

```diff
@@ -0,0 +1,96 @@
+import type { ChatCompletionCreateParamsBase } from "openai/resources/chat/completions";
+
+export const convertFunctionCall = (
+  functionCall: ChatCompletionCreateParamsBase["function_call"] | null,
+): ChatCompletionCreateParamsBase["tool_choice"] => {
+  if (!functionCall) {
+    return undefined;
+  } else if (functionCall === "auto" || functionCall === "none") {
+    return functionCall;
+  } else if (functionCall.name) {
+    return {
+      type: "function",
+      function: {
+        name: functionCall.name,
+      },
+    };
+  }
+};
+
+export const convertFunctions = (
+  functions: ChatCompletionCreateParamsBase["functions"] | null,
+): ChatCompletionCreateParamsBase["tools"] => {
+  if (!functions) return undefined;
+  return functions.map((func) => ({
+    type: "function",
+    function: {
+      name: func.name,
+      description: func.description,
+      parameters: func.parameters,
+    },
+  }));
+};
+
+export const convertMessages = (
+  messages: ChatCompletionCreateParamsBase["messages"],
+): ChatCompletionCreateParamsBase["messages"] =>
+  messages ? messages.map(convertFunctionMessageToToolCall) : [];
```

Comment:
Can messages ever be null? If not, this check isn't necessary. And if it can be, we should just return the null, not coerce it to an empty array.

Reply:
No they can't. This was left over from an earlier refactor. Good catch.